src/OSPSuite.InstallationValidator.Core/Services/LogWatcherFactory.cs

```diff
@@ -0,0 +1,9 @@
+﻿using OSPSuite.InstallationValidator.Core.Domain;
+
+namespace OSPSuite.InstallationValidator.Core.Services
+{
+   public interface ILogWatcherFactory
+   {
+      LogWatcher CreateLogWatcher(string logFile);
```

Comment:
That;s pretty neat. Castle seems to pass all parameters from the method to the constructor and then also add other parameters such as EventPublisher. Did not know that